Deprecation warning when compiling finalize method in Java 9

I'm using Swig 3.0.12, I've recently updated my java installation:
```shell
$ java --version
openjdk 9.0.4
OpenJDK Runtime Environment (build 9.0.4+12-Debian-4)
OpenJDK 64-Bit Server VM (build 9.0.4+12-Debian-4, mixed mode)
```
and I get the following output when compiling the Java code generated by Swig:
```
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```
```
...
/some/java/class/generated/by/swig/MyClass.java:44: warning: [deprecation] finalize() in Object has been deprecated
  protected void finalize() {
                 ^
...
Note: Some input files additionally use or override a deprecated API.
100 warnings
```

Manually adding `@SuppressWarnings("deprecation")` to all the `finalize` methods seems to work. Unfortunately this is not a solution as this is code generated by swig.

I don't know what's the best approach for swig here (generate the suppression annotation or find a replacement for `finalize`) as I'm not a Java expert.

In the mean time, does one see a workaround that could allow me to silence these warnings without having to manually patch the generated code each time I call swig?

--- a/Examples/test-suite/java_director.i
+++ b/Examples/test-suite/java_director.i
@@ -7,6 +7,7 @@
 %module(directors="1") java_director
 
 %typemap(javafinalize) SWIGTYPE %{
+  @SuppressWarnings("deprecation")
   protected void finalize() {
 //    System.out.println("Finalizing " + this);
     delete();

--- a/Examples/test-suite/java_throws.i
+++ b/Examples/test-suite/java_throws.i
@@ -192,6 +192,7 @@ try {
 
 // Need to handle the checked exception in NoExceptTest.delete()
 %typemap(javafinalize) SWIGTYPE %{
+  @SuppressWarnings("deprecation")
   protected void finalize() {
     try {
       delete();

--- a/Lib/java/java.swg
+++ b/Lib/java/java.swg
@@ -1273,6 +1273,7 @@ SWIG_JAVABODY_PROXY(protected, protected, SWIGTYPE)
 SWIG_JAVABODY_TYPEWRAPPER(protected, protected, protected, SWIGTYPE)
 
 %typemap(javafinalize) SWIGTYPE %{
+  @SuppressWarnings("deprecation")
   protected void finalize() {
     delete();
   }

